### PR TITLE
docs: add docs

### PR DESCRIPTION
**Pull request by nsakaimbo**
### Description

Add (and update) desktop docs that weren't part of the initial migration from wp-desktop repo.

### CODE REVIEW

**Review by belcherj — APPROVED** [review]
Did not follow all of these flows but the text reads fine. I will follow all the flows over the coming weeks.

File: desktop/docs/development.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+## Starting the app
+
+### How does it work?
+
+It's a fairly complicated process so buckle up. Note that *(main)* and *(renderer)* will be added to show where the code actually runs.
+
+For clarity, all file and folder locations are relative to the root of the Calypso monorepo.
+
+- *(main)* Electron looks at the `main` item in `desktop/package.json` - this is the boot file, and refers to `client/desktop/index.js`
```
Comment (jsnajdr) [inline desktop/docs/development.md:22]: The `main` item refers to `build/desktop.js`, which is the entrypoint of the compiled webpack bundle for the server running in Electron Main Process.

`client/desktop/index.js` is the root source file (entry) where the webpack compilation starts from.

File: desktop/docs/development.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+- *(main)* `client/desktop/index.js` sets up the environment in `client/desktop/env.js` - this includes Node paths for Calypso
+- *(main)* Various [app handlers](../../client/desktop/app-handlers/README.md) are loaded from `client/desktop/app-handlers` - these are bits of code that run before the main window opens
+- *(main)* A Calypso server is started drectly from Electron's Node process in `client/desktop/server.js`. The server is customized to serve files from the following directories:
```
Comment (jsnajdr) [inline desktop/docs/development.md:25]: It'd be good to note here that the "Calypso" server is an Express.js HTTP server that serves files for the Renderer process.

File: desktop/docs/development.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  - `/` - mapped to the Calypso server
```
Comment (jsnajdr) [inline desktop/docs/development.md:26]: I think the `/` path is no longer mapped. We serve only the `/calypso` directories (mapped to the `/public` dir on filesystem) and `/desktop`, mapped to `/public_desktop`.

File: desktop/docs/development.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  - `/calypso` - mapped to `calypso/public`
+  - `/desktop` - mapped to `public_desktop`
+- *(main)* An Electron `BrowserWindow` is opened and loads the 'index' page from the Calypso server
+- *(main)* Once the window has opened the [window handlers](../../client/desktop/window-handlers/README.md) load to provide interaction between Calypso and Electron
+- *(renderer)* Calypso provides the 'index' page from `calypso/server/pages/desktop.pug`, which is a standard Calypso start page plus:
```
Comment (jsnajdr) [inline desktop/docs/development.md:31]: We no longer use `*.pug` files today. The index page is served by a React renderer in `client/document/desktop.jsx`.

File: desktop/docs/development.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  - `desktop/public_desktop/wordpress-desktop.css` - any CSS specific to the desktop app
+  - `desktop/public_desktop/desktop-app.js` - desktop app specific JS and also the Calypso boot code
```
Comment (jsnajdr) [inline desktop/docs/development.md:33]: These two files are part of the `/desktop` directory mentioned above.

File: desktop/docs/development.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  - `desktop/calypso/public/build.js` - a prebuilt Calypso
```
Comment (jsnajdr) [inline desktop/docs/development.md:34]: The Calypso client bundle for desktop is written to `desktop/public` during the build. And its entrypoint is not `build.js`, but it has multiple webpack entrypoints (like `entry-main.[hash].min.js`). The file names are written to `desktop/build/assets-evergreen.js` during the build, and the Express.js server loads that file to figure out which `<script>` and `<link rel="stylesheet">` tags to send to the browser.

**Pull request merged**